code/modules/antagonists/traitor/contractor/datums/syndicate_contract.dm

```diff
@@ -0,0 +1,518 @@
+/datum/syndicate_contract/proc/fail(reason = "")
+	if(status != CONTRACT_STATUS_ACTIVE)
+		return
+
+	// Update info
+	owning_hub.current_contract = null
+	status = CONTRACT_STATUS_FAILED
+	fail_reason = reason
+	// Notify
+	clean_up()
+	owning_hub.uplink?.message_holder("You failed to kidnap the target, agent. Do not disappoint us again.", 'sound/machines/terminal_prompt_deny.ogg')
+
+/**
+  * Initiates the extraction process if conditions are met.
+  *
+  * Arguments:
+  * * M - The contractor.
+  */
+/datum/syndicate_contract/proc/start_extraction_process(obj/item/contractor_uplink/U, mob/living/carbon/human/M)
+	if(!U?.Adjacent(M))
+		return "Where in space is your uplink?!"
+	else if(status != CONTRACT_STATUS_ACTIVE)
+		return "This contract is not active."
+	else if(extraction_deadline > world.time)
+		return "Another extraction attempt cannot be made yet."
+
+	var/mob/target = contract.target.current
+	if(!target)
+		invalidate()
+		return "The target is no longer on our sensors. Your contract will be invalidated and replaced with another one."
+	else if(!contract.can_start_extraction_process(M, target))
+		return "You and the target must be standing in the extraction area to start the extraction process."
+
+	M.visible_message("<span class='notice'>[M] starts entering a cryptic series of characters on [U].</span>",\
+					  "<span class='notice'>You start entering an extraction signal to your handlers on [U]..</span>")
+	if(do_after(M, EXTRACTION_PHASE_PREPARE, target = M))
+		if(!U?.Adjacent(M) || extraction_deadline > world.time)
+			return
+		var/obj/effect/contractor_flare/F = new(get_turf(M))
+		extraction_flare = F
+		extraction_deadline = world.time + extraction_cooldown
+		M.visible_message("<span class='notice'>[M] enters a mysterious code on [U] and pulls a black and gold flare from [M.p_their()] belongings before lighting it.</span>",\
+						  "<span class='notice'>You finish entering the signal on [U] and light an extraction flare, initiating the extraction process.</span>")
+		addtimer(CALLBACK(src, .proc/extraction_step_1, U, M, F), EXTRACTION_PHASE_STEP_1)
+		extraction_timer_handle = addtimer(CALLBACK(src, .proc/deadline_reached), portal_duration, TIMER_STOPPABLE)
+
+/**
+  * First step of the extraction process.
+  *
+  * Arguments:
+  * * U - The uplink.
+  * * M - The contractor.
+  * * F - The flare.
+  */
+/datum/syndicate_contract/proc/extraction_step_1(obj/item/contractor_uplink/U, mob/living/carbon/human/M, obj/effect/contractor_flare/F)
+	if(!U || !M || status != CONTRACT_STATUS_ACTIVE)
+		invalidate()
+		return
+	else if(!F)
+		U.message_holder("Extraction flare could not be located, agent. Ensure the extraction zone is clear before signaling us.", 'sound/machines/terminal_prompt_deny.ogg')
+		return
+	else if(!ismob(contract.target?.current))
+		invalidate()
+		return
+	U.message_holder("Extraction signal received, agent. [GLOB.using_map.full_name]'s redspace transport jamming systems have been sabotaged. "\
```

Comment:
Lore thing (though lore is mostly dead). I believe redspace was the evil untamed *magic* where things like slaughter demons and cult gods come from. Bluespace being the technology variant of magic